### PR TITLE
CI: link lint + spell check

### LINKED ISSUE #178: add broken link check CI
ref: #176
https://github.com/tcort/markdown-link-check on every push?

### PR DESCRIPTION
**Pull request by Reecepbcups**
closes #178

Fixes all broken links and common spelling issues

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,12 +74,13 @@ An [example IBC app](./examples/hello-world/)
...
 | [query](https://github.com/defund-labs/defund/tree/main/x/query) | Module | [Defund Labs](https://github.com/defund-labs) | An application that enables on chain querying of another IBC enabled chains state without the need for the chain being queried to implement the application. Similar to the interchain-queries application in the row above but without callbacks. |
 | [NFT Transfer (ICS 721)](https://github.com/bianjieai/nft-transfer) | Module | [Bianjieai](https://github.com/bianjieai) | An application that enables cross chain NFT transfer. |
-| [CosmWasm NFT Transfer (ICS 721)](https://github.com/public-awesome/cw-ics721) | WASM Contract | [Public Awesome (Stargaze)](https://github.com/public-awesome), [Ark Protocol](https://twitter.com/ArkProtocol) | An application that enables cross chain NFT transfer. CosmWasm implementation of the above, written in Rust. |
-| [recovery](https://github.com/evmos/evmos/tree/main/x/recovery) | Middleware | [Evmos](https://github.com/evmos) | Middleware enabling the recovery of tokens sent to unsupported addresses. |
+<!-- markdown-link-check-disable-next-line -->
```
Comment (Reecepbcups) [inline README.md:81]: x.com throws issues bleh

File: .codespellrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[codespell]
+skip = *.pulsar.go,*.pb.go,*.pb.gw.go,*.cosmos_orm.go,*.json,*.git,*.js,crypto/keys,fuzz,*.h,proto/tendermint,*.bin,*.sum,*.mod,*_test.go
```
Comment (jtieri) [inline .codespellrc:2]: i notice that here we skip Go test files but your PR in the relayer didn't, is this on purpose?

File: .github/markdown-link-check-config.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "timeout": "20s",
+    "retryOn429": true,
+    "retryCount": 5,
+    "fallbackRetryDelay": "30s",
+    "aliveStatusCodes": [
+        200,
+        206
+    ]
+}
```
Comment (jtieri) [inline .github/markdown-link-check-config.json:10]: why the need for this config file in this repo but not in the relayer repo?

**Review by jtieri — COMMENTED** [review]
had a couple questions but otherwise this lgtm

File: .github/markdown-link-check-config.json (hunk 3 above)
Comment (Reecepbcups) [reply .github/markdown-link-check-config.json:10]: There are more links in this repo and some edge case URLs being used. Given the higher failure rate the config helps with some flakey URLs (cough cough twittter)

File: .codespellrc (hunk 2 above)
Comment (Reecepbcups) [reply .codespellrc:2]: yes, many test files use things like userA, step, and other words which are commonly flagged as incorrectly spelled. With the different cases it messes with codespell and is kind of a pain. so just ignoring for now

**Review by jtieri — APPROVED** [review]

**Comment by jtieri** [comment]
If it's not any trouble could we backport this to the other maintained branches? If you don't have any extra bandwidth we could just open an issue as a reminder and get someone else assigned to it at some point.

**Pull request merged**